src/internet_identity/src/storage/anchor.rs

```diff
@@ -369,8 +397,8 @@ fn check_anchor_invariants(devices: &Vec<&Device>) -> Result<(), AnchorError> {
 
     /// Single devices can use >500 bytes for the variable length fields alone.
     /// In order to not give away all the anchor space to the device vector, we limit the sum of the
-    /// size of all variable fields of all devices. This ensures that we have the flexibility to expand
-    /// or change anchors in the future.
+    /// size of all variable fields of all devices plus the device independent metadata.
+    /// This ensures that we have the flexibility to expand or change anchors in the future.
     /// The value 2500 was chosen so to accommodate pre-memory-migration anchors (limited to 2048 bytes)
     /// plus an additional 452 bytes to fit new fields introduced since.
     const VARIABLE_FIELDS_LIMIT: usize = 2500;
```

Comment:
It seems that the metadata needs a separate quota in case the devices already use up the full 2500 and the frontend needs to put some metadata.